docs/bills-matching.md

```diff
@@ -0,0 +1,139 @@
+the ones that fits in the amount and date windows of the bill.
+
+The date window is `[bill date - dateLowerDelta; bill date + dateUpperDelta]`. Where:
+
+* `bill date` is `bill.date` if we are looking for a credit; `bill.originalDate` with a fallback on `bill.date` if we are looking for a debit
+* `dateLowerDelta` is `bill.matchingCriterias.dateLowerDelta` if it exists, `options.pastWindow` otherwise
+* `dateUpperDelta` is `bill.matchingCriterias.dateUpperDelta` if it exists, `options.futureWindow` otherwise
+
+The amount window is `[bill amount - amountLowerDelta; bill amount + amountUpperDelta]`. Where:
+
+* `bill amount` is `bill.groupAmount` with a fallback on `bill.amount` if we are looking for a credit; `-bill.originalAmount` with a fallback on `-bill.amount` if we are looking for a debit
+* `amountLowerDelta` is `bill.matchingCriterias.amountLowerDelta` if it exists, `options.minAmoutDelta` otherwise
+* `amountUpperDelta` is `bill.matchingCriterias.amountUpperDelta` if it exists, `options.maxAmoutDelta` otherwise
+
+All transactions that matches these two criterias are kept.
+
+### Apply filters to neighboring transactions
+
+There are other filters a transaction must pass to match with a bill.
+
+#### Category
+
+Health bills are a little special because they can only match with health
+transactions. So if the bill is an health bill, we keep only health
+transactions. Otherwise, we keep all but health transactions.
+
+You can pass a `allowUncategorized` option to the algorithm, in which case
+uncategorized transactions will also be able to match bills.
+
+#### Reimbursements
+
+This filter is applied only if we are looking for a debit transactions.
+
+If we try to link a bill as a reimbursement to a debit transaction, we check
+that the bill's amount and transaction's reimbursements sum does not overflows
+the transaction's amount.
+
+#### Label regex
+
+This filter is applied only if we are looking for a credit transaction or if
+the bill is not of health type.
+
+This filter checks if a transaction matches a regex that is built upon the
+bill's properties. Here is how the regexp is built:
+
+* If the bill has a `matchingCriterias.labelRegex` property, this is used
+* Otherwise we try to get the brand corresponding to the bill's vendor in our [brands dictionary](https://github.com/cozy/cozy-banks/blob/master/src/ducks/brandDictionary/brands.json). If it exists, we use its `regexp` property
+* Otherwise, we create a regexp containing the bill's vendor
+* Lastly, if the bill has no vendor, we can't do anything and the filter will be `false` for every transaction and the bill will not match any transaction
+
+### Sort remaining transactions
+
+A score is given to the remaining transactions according to their « distance »
+to the bill's date and amount. The closest one wins and continues through the
+algorithm.
+
+### Link bills to debit and credit transactions
+
+When transactions were found for a given bill, we need to link the bill to
```

Comment:
```suggestion
When transactions are found for a given bill, we need to link the bill to
```

I think the past made it sound like "if transactions were found for a given bill in the past and not during the course of the matching".